src/test/java/com/thirdcc/webapp/web/rest/EventActivityResourceIT.java

```diff
@@ -110,52 +123,205 @@ public static EventActivity createEntity(EntityManager em) {
+
+        restEventActivityMockMvc.perform(post("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeCreate);
+    }
+
+    @Test
+    public void createEventActivity_WithEventCancelled_ShouldThrow400() throws Exception {
+        event.setStatus(EventStatus.CANCELLED);
+        Event savedEvent = initEventDB();
+
+        int databaseSizeBeforeCreate = eventActivityRepository.findAll().size();
+
+        assertThat(savedEvent.getStatus()).isEqualByComparingTo(EventStatus.CANCELLED);
+
+        EventActivityDTO eventActivityDTO = createDefaultEventActivityDTO();
+        eventActivityDTO.setEventId(savedEvent.getId());
+
+        restEventActivityMockMvc.perform(post("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeCreate);
+    }
+
+    @Test
+    public void createEventActivity_WithStartDateEarlierThanToday_ShouldThrow400() throws Exception {
+        Event savedEvent = initEventDB();
+
+        int databaseSizeBeforeCreate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createDefaultEventActivityDTO();
+        eventActivityDTO.setEventId(savedEvent.getId());
+        eventActivityDTO.setStartDate(Instant.now().minus(1, ChronoUnit.SECONDS));
+
+        assertThat(eventActivityDTO.getStartDate()).isBefore(Instant.now());
+
+        restEventActivityMockMvc.perform(post("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeCreate);
+    }
+
+    @Test
+    public void createEventActivity_WithStartDateEarlierThanEventStartDate_ShouldThrow400() throws Exception {
+        event.setStartDate(Instant.now().plus(1, ChronoUnit.DAYS));
+        Event savedEvent = initEventDB();
+
+        int databaseSizeBeforeCreate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createDefaultEventActivityDTO();
+        eventActivityDTO.setEventId(savedEvent.getId());
+        eventActivityDTO.setStartDate(savedEvent.getStartDate().minus(1, ChronoUnit.SECONDS));
+
+        assertThat(savedEvent.getStartDate()).isAfter(Instant.now()); //for more accurate testing
```

Comment:
Why this line is only appear in one test case?

Reply:
the original motivation is to ensure line 68 of EventActivityServiceImpl.java
`if (eventActivityDTO.getStartDate().isBefore(event.getStartDate()))`
 is tested.

But line 65 of EventActivityServiceImpl.java 
`if (eventActivityDTO.getStartDate().isBefore(Instant.now()))`
may return true, and proceed to throw error

so this line (line 291 EventActivityResourceIT.java) was added
`assertThat(savedEvent.getStartDate()).isAfter(Instant.now()); //for more accurate testing`
to ensure that we are really testing line 68 of EventActivityServiceImpl.java

**But** this test case is removed in f8fd7839e49f49048eb7115c741ac59c12452a2c. Due to a business logic change. Refer #7